flang/lib/Optimizer/Dialect/FIROps.cpp

```diff
@@ -1787,6 +1840,10 @@ llvm::LogicalResult fir::EmboxOp::verify() {
     return emitOpError("slice must not be provided for a scalar");
   if (getSourceBox() && !mlir::isa<fir::ClassType>(getResult().getType()))
     return emitOpError("source_box must be used with fir.class result type");
+  // if (fir::isa_volatile_type(getMemref().getType()) !=
+  //     fir::isa_volatile_type(getResult().getType()))
+  //   return emitOpError("input and output types must have the same
+  //   volatility");
```

Comment:
```suggestion
```

Reply:
I actually meant to keep this check (and use it generally on ops that might take a volatile ref/box) to make sure ops aren't accidentally dropping volatility - what do you think about this?